Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean

```diff
@@ -566,6 +566,24 @@ theorem measure_mono_null_ae (H : s ≤ᵐ[μ] t) (ht : μ t = 0) : μ s = 0 :=
   nonpos_iff_eq_zero.1 <| ht ▸ H.measure_le
 #align measure_theory.measure_mono_null_ae MeasureTheory.measure_mono_null_ae
 
+/-- Two functions are almost everywhere equal on their entire domain iff they are
+a.e equal on every subsets of their domain. -/
+theorem fun_ae_imp_set_ae {f g : α → β} :
+    f =ᵐ[μ] g ↔ ∀ (s : Set α), ∀ᵐ x ∂ μ, x ∈ s → f x = g x :=
+  Iff.intro
+    (fun h s ↦ Eventually.mono h fun x a _ ↦ a)
+    (fun h ↦ by
+      let A := {x | f x = g x}
+      let B := {x | x ∈ Set.univ → f x = g x}
+      have B_eq_A : B = A := by {
+        ext e
+        exact Iff.intro (fun hb ↦ hb (by simp)) (fun ha _ ↦ ha)
+      }
+      specialize h Set.univ
+      unfold Filter.Eventually at *
+      rwa [show {x | (fun x ↦ x ∈ univ → f x = g x) x} = B by rfl, B_eq_A] at h
+    )
```

Comment:
```suggestion
    (fun h ↦ by filter_upwards [h univ] with x hx using by simpa using hx)
```
but to be honest, I am not really convinced by the lemma: do you have an example where having the lemma is more convenient than a `filter_upwards` application?